Taskfile.yml

```diff
@@ -111,17 +112,26 @@ tasks:
       - pug -w -P --doctype html $(find ./html/ -name "*.pug") &
 
   build:
-    desc: Build a full set of release binaries
-    dir: cli
+    desc: Build a full set of release binaries and packages
     cmds:
-      - gox -output="../bin/semaphore_{{"{{"}}.OS{{"}}"}}_{{"{{"}}.Arch{{"}}"}}" ./...
+      - task: release:test
 
   build:local:
     desc: Build a binary for the current architecture
     dir: cli
     cmds:
      - go build -o ../bin/semaphore
 
+  release:test:
+    desc: creates a release without performing validations or publishing artifacts
+    cmds:
+      - goreleaser --snapshot --rm-dist
+
+  release:
+    desc: Build a full set of release binaries
+    cmds:
+      - goreleaser
```

Comment:
`goreleaser --rm-dist` is better in case if we make the release on the developer's machine.

```
$ task release
goreleaser

   • releasing using goreleaser dev...
   • loading config file       file=.goreleaser.yml
   • SETTING DEFAULTS FOR:
      • loading environment variables
      • snapshoting              
      • releasing to GitHub      
      • project name             
      • creating archives        
      • building binaries        
      • creating Linux packages with fpm
      • creating Linux packages with nfpm
      • creating Linux packages with snapcraft
      • calculating checksums    
      • signing artifacts        
      • creating Docker images   
      • releasing to Artifactory 
      • creating homebrew formula
      • creating Scoop Manifest  
   • CHECKING ./DIST  
   ⨯ release failed after 0.00s error=bin is not empty, remove it before running goreleaser or use the --rm-dist flag
task: Failed to run task "release": exit status 1
```

Reply:
a real release, with the changelogs etc.... should never be run on a developers machine, that should be only ever done in the ci (imo) as it potentially pushes changes, and it requires a totally clean git tree. In reality it means that only us guys (or the ci) can do it anyway, as no one else is allowed to create releases, but i'm happy to add that flag if you want

Reply:
Maybe should add some warning then? Like "NEVER RUN THIS TASK ON DEVELOPER MACHINE" :)

Reply:
well it should just fail on a developer machine as they cant push it and the copy will most probably be dirty, so nothing can get totally fucked up from running it, unless one of us forgets about it and runs it. 
However I might remove the description as currently all other commands that people are not meant to call dont have them and as such are hidden from the task list `task -l` or `task` output.
Any better solution you can think of?

Reply:
>unless one of us forgets about it and runs it
If shit can happen - it will happen. Murphy's law.

Removing description is good for me. Another thought - rename `release:test` to `release` and `release` to `release:prod` (or so). Frequently used commands should be shorter than rarely used (and potentially dangerous) commands.

Reply:
yeah i like that approach. I'll remove description and change the task names to make `release` the safe one